### PR TITLE
Feature/milestone 5

### PR DESCRIPTION
**Pull request by Fig77**
* Added models friend request.
* User can request friendship.
* User can see if someone requested a friendship.
* User can accept or reject said friendship.
* These things can be seen either on the user page or when displaying all users.

### CODE REVIEW

File: app/models/friend_request.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class FriendRequest < ApplicationRecord
```
Comment (mcrd25) [inline app/models/friend_request.rb]: It's cool if you want to name your friendship model friend request. But please remember this article https://smartfunnycool.com/friendships-in-activerecord/ on how to implement this model successfully.

File: app/models/user.rb [hunk 2]
```diff
@@ -9,4 +9,5 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friend_requests, foreign_key: :creator_id
```
Comment (mcrd25) [inline app/models/user.rb]: You are missing another association for inverted friendship. Please see this article: https://smartfunnycool.com/friendships-in-activerecord/

I.e what about the friendship association for someone  that is not creator `friend_id`

File: app/models/friend_request.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  belongs_to :creator, class_name: 'User'
```
Comment (mcrd25) [inline app/models/friend_request.rb]: missing belongs_to association for a friend. Again, please this article for a hint:  https://smartfunnycool.com/friendships-in-activerecord/

**Comment by samgaco** [comment]
Hi @Dandush03 and @Fig77 :high_brightness: 

I have checked you implemented all the required changes in the previous review, you are now good to move on :+1: 

-------------------------

* ###### STATUS OF THE MILESTONE: APPROVED :heavy_check_mark:

**Pull request merged**